### ISSUE TITLE
Improper reusing horizontal - False Positive

### ISSUE THREAD

**Issue by RickBrice** [buildingSMART/ifc-gherkin-rules#442]
The attached example illustrates an improper use of [CT 4.1.4.4.1.1](https://ifc43-docs.standards.buildingsmart.org/IFC/RELEASE/IFC4x3/HTML/concepts/Object_Composition/Nesting/Alignment_Layouts/Alignment_Layout_-_Horizontal,_Vertical_and_Cant/content.html). The alignment contains two vertical layouts and the validation service does not flag it as an error.

The structure is:
IfcAlignment-IfcRelNests-IfcReferent,IfcAlignmentHorizontal,IfcAlignmentVertical,IfcAlignmentVertical.

Reusing a horizontal with two vertical is governed by [CT 4.1.4.4.1.2](https://ifc43-docs.standards.buildingsmart.org/IFC/RELEASE/IFC4x3/HTML/concepts/Object_Composition/Nesting/Alignment_Layouts/Alignment_Layout_-_Reusing_Horizontal_Layout/content.html)

[Improper_Reusing_Horizontal.ifc.txt](https://github.com/user-attachments/files/21953530/Improper_Reusing_Horizontal.ifc.txt)

**Comment by civilx64** [comment]
So, something like:

```
Given an .IfcAlignment.
Given its attribute .IsNestedBy.
Then there must be ^no more than one^ instance of .IfcAlignmentVertical.
```

**Comment by RickBrice** [comment]
and ^no more than one^ instance of IfcAlignmentCant

alternatively

Given an IfcAlignment
Given its attribute IfcNestedBy
There must be only one IfcAlignmentHorizontal, or one IfcAlignmentHorizontal and one IfcAlignmentVertical, or one IfcAlignmentHorizontal, one IfcAlignmentVertical, and one IfcAlignmentCant,

Not sure if one horizontal + one cant is valid given the requirements for geometric representation from CT 4.1.7.1.1.3 where IfcGradientCurve is the BaseCurve for IfcSegmentedReferenceCurve. The gradient curve implies cant lays on top of vertical.

**Comment by civilx64** [comment]
You are correct, of course.  The rule needs to flag H + C with no V as invalid.

**Comment by civilx64** [comment]
@RickBrice update: I have a working feature file to address the first CT, which will be new rule ALB031.  I also have started a draft of another new rule ALB032 to enforce the second CT.  

Do you have (or could quickly generate) any additional test data for valid and invalid permutations of layout nesting?  For example, 2 horiz + 1 vert + 2 cant all in basic nesting, horizontal re-use with a horizontal incorrectly decomposing, etc.  Hopefully that makes sense.  Current WIP is in branch [IVS-620](https://github.com/buildingSMART/ifc-gherkin-rules/tree/IVS-620-improper-reusing-horizontal).
